### PR TITLE
added private message reporting, blocking user from comment

### PR DESCRIPTION
**Pull request by EricBAndrews**
Added backend support for reporting private messages and added a report option to the private message context menu.

Added support for blocking a user from all inbox items.

Additionally added a block user function to comments, plus a filtering function on commentTracker to remove blocked users.

### CODE REVIEW

File: Mlem/Logic/Participating/BlockPersonLogic.swift [hunk 1]
```diff
@@ -20,3 +20,17 @@ func blockPerson(
...
+    account: SavedAccount,
+    personId: Int,
+    blocked: Bool) async throws -> Bool {
+        let request = BlockPersonRequest(
+            account: account,
+            personId: personId,
+            block: blocked
+        )
+        let response = try await APIClient().perform(request: request)
+        return response.blocked
```
[the remark this answers is not in the document]
Comment (EricBAndrews) [reply Mlem/Logic/Participating/BlockPersonLogic.swift:35]: Once we have unified haptics, yes--not worth until then IMO, writing DOA code

File: Mlem/Models/Trackers/Comment Tracker.swift [hunk 2]
```diff
@@ -18,4 +18,24 @@ class CommentTracker: ObservableObject {
         let accepted = newComments.filter { ids.insert($0.id).inserted }
         comments.append(contentsOf: accepted)
     }
+    
+    // Takes a callback and fillters out any entry that returns false
+    //
+    // Returns the number of entries removed
+    @discardableResult func filter(_ callback: (HierarchicalComment) -> Bool) -> Int {
+        var removedElements = 0
```
[the remark this answers is not in the document]
Comment (EricBAndrews) [reply Mlem/Models/Trackers/Comment Tracker.swift:26]: Opened #297

**Comment by EricBAndrews** [comment]
Moved everything that is a "response to a thing"--comments, messages, reports--to use a general-purpose ResponseComposer.

File: Mlem/Views/Tabs/Inbox/Feed/InteractionSwipeAndMenuHelpers.swift [hunk 3]
```diff
@@ -80,6 +81,24 @@ extension InboxView {
...
+        ret.append(MenuFunction(text: "Report Comment",
+                                imageName: AppConstants.reportSymbolName,
+                                destructiveActionPrompt: nil,
+                                enabled: true) {
+            reportCommentReply(commentReply: commentReply)
+        })
+        
+        // block
+        ret.append(MenuFunction(text: "Block User",
+                                imageName: "person.fill.xmark",
```
[the remark this answers is not in the document]
Comment (EricBAndrews) [reply Mlem/Views/Tabs/Inbox/Feed/InteractionSwipeAndMenuHelpers.swift]: Nice catch!

**Review by WestonHanners — APPROVED** [review]

**Pull request merged**